### PR TITLE
[cu-28fuhkc] Subsquid Unit Tests + LiquidityRemoved & PoolDeleted events processing 

### PR DESCRIPTION
**Pull request by vimukthi-git**
## Issue
https://app.clickup.com/t/28fuhkc

## Description
Subsquid Happy path Unit Tests + LiquidityRemoved & PoolDeleted events processing. 

## Checklist

- [x] I have updated the cargo docs to reflect changes made by this PR _(if applicable)_
- [x] I have updated the `book/` to reflect changes made by this PR _(if applicable)_

### CODE REVIEW

**Review by dzmitry-lahoda — APPROVED** [review]

File: subsquid/src/pabloProcessor.ts [hunk 1]
```diff
@@ -188,6 +197,81 @@ export async function processLiquidityAddedEvent(ctx: EventHandlerContext, event
...
+        const {who, poolId, baseAmount, quoteAmount, totalIssuance} = event.asV2100;
+        return {who, poolId, baseAmount, quoteAmount, totalIssuance};
+    } else {
+        const {who, poolId, baseAmount, quoteAmount, totalIssuance} = event.asLatest;
+        return {who, poolId, baseAmount, quoteAmount, totalIssuance};
+    }
+}
+
+export async function processLiquidityRemovedEvent(ctx: EventHandlerContext, event: PabloLiquidityRemovedEvent) {
+    console.debug('processing LiquidityAddedEvent', event);
```
Comment (vivekvpandya) [inline subsquid/src/pabloProcessor.ts:219]: confusing log.

File: subsquid/src/pabloProcessor.ts [hunk 2]
```diff
@@ -279,10 +363,82 @@ export async function processSwappedEvent(ctx: EventHandlerContext, event: Pablo
...
+        const {poolId, baseAmount, quoteAmount} = event.asV2100;
+        return {poolId, baseAmount, quoteAmount};
+    } else {
+        const {poolId, baseAmount, quoteAmount} = event.asLatest;
+        return {poolId, baseAmount, quoteAmount};
+    }
+}
+
+export async function processPoolDeletedEvent(ctx: EventHandlerContext, event: PabloPoolDeletedEvent) {
+    console.debug('processing LiquidityAddedEvent', event);
```
Comment (vivekvpandya) [inline subsquid/src/pabloProcessor.ts:399]: confusing log.

File: subsquid/src/pabloProcessor.ts [hunk 3]
```diff
@@ -279,10 +363,82 @@ export async function processSwappedEvent(ctx: EventHandlerContext, event: Pablo
...
+
+        let tx = await get(ctx.store, PabloTransaction, ctx.event.id);
+        if (tx != undefined) {
+            throw new Error("Unexpected transaction in db");
+        }
+        tx = createTransaction(ctx, pool, who,
+            PabloTransactionType.DELETE_POOL,
+            Big(poolDeletedEvent.baseAmount.toString())
+                .div(Big(poolDeletedEvent.quoteAmount.toString())).toString(),
+            BigInt(baseAsset.assetId),
```
Comment (vivekvpandya) [inline subsquid/src/pabloProcessor.ts:438]: may just use pool.baseAssetId just to make it consistent, or I am missing something?

Comment (vimukthi-git) [reply subsquid/src/pabloProcessor.ts:438]: No pool does not have a `baseAssetId`. The field is tracking the default base for tracking stats, but does not assume a default base asset Id for forward compatibility with multi asset pools

**Review by hussein-aitlahcen — DISMISSED** [review]

**Review by vivekvpandya — APPROVED** [review]

**Review by dzmitry-lahoda — APPROVED** [review]

**Pull request merged**